### PR TITLE
Cryptocurrency searching filter 

### PR DESCRIPTION
**Pull request by DianaCrainic**

### CODE REVIEW

File: cryptocurrencies-service/src/services/cryptocurrency-service.ts [hunk 1]
```diff
@@ -117,13 +120,17 @@ export const getCryptocurrencies = async (limit = 10, offset = 0): Promise<Crypt
     return result["@graph"] as Cryptocurrency[];
 };
 
-export const getCryptocurrenciesInfo = async (): Promise<CryptocurrenciesInfo> => {
+export const getCryptocurrenciesInfo = async (searchText = [""]): Promise<CryptocurrenciesInfo> => {
```
Comment (IoanSava) [inline cryptocurrencies-service/src/services/cryptocurrency-service.ts]:
```
export const getCryptocurrenciesInfo = async (searchText: string[] = [""]): Promise<CryptocurrenciesInfo> => {
```

File: cryptocurrencies-service/src/services/cryptocurrency-service.ts [hunk 2]
```diff
@@ -66,7 +66,9 @@ export const getCryptocurrencyById = async (id: string): Promise<any> => {
     return cryptocurrency;
 }
 
-export const getCryptocurrencies = async (limit = 10, offset = 0): Promise<Cryptocurrency[]> => {
+export const getCryptocurrencies = async (limit = 10, offset = 0, searchText = [""]): Promise<Cryptocurrency[]> => {
+    const whereClause = searchText.map(text => `regex(?symbol, "${text}", "i")`).join(" || ");
```
Comment (IoanSava) [inline cryptocurrencies-service/src/services/cryptocurrency-service.ts]: Rename this variable and use the same statements we used within `getCryptocurrenciesInfo` (`CONTAINS`, `LCASE`).

File: cryptocurrencies-service/src/services/cryptocurrency-service.ts [hunk 3]
```diff
@@ -66,7 +66,9 @@ export const getCryptocurrencyById = async (id: string): Promise<any> => {
     return cryptocurrency;
 }
 
-export const getCryptocurrencies = async (limit = 10, offset = 0): Promise<Cryptocurrency[]> => {
+export const getCryptocurrencies = async (limit = 10, offset = 0, searchText = [""]): Promise<Cryptocurrency[]> => {
```
Comment (IoanSava) [inline cryptocurrencies-service/src/services/cryptocurrency-service.ts]: `searchText: string[] = [""]`

File: cryk-ui/src/pages/Cryptocurrencies.tsx [hunk 4]
```diff
@@ -178,6 +269,35 @@ export default function Cryptocurrencies() {
...
+                        <div className="search-box">
+                            <Search>
+                                <SearchIconWrapper>
+                                    <SearchIcon />
+                                </SearchIconWrapper>
+                                <StyledInputBase
+                                    placeholder="ace,bit..."
+                                    inputProps={{ 'aria-label': 'search' }}
+                                    value={value}
+                                    onChange={e => setValue(e.target.value)}
```
Comment (IoanSava) [inline cryk-ui/src/pages/Cryptocurrencies.tsx]: `e` => `event`

File: cryk-ui/src/pages/Cryptocurrencies.tsx [hunk 5]
```diff
@@ -106,16 +112,71 @@ const getBarChart = (data: { name: string, value: number }[]): any => {
     </ResponsiveContainer>);
 }
 
+const Search = styled('div')(({ theme }) => ({
+    position: 'relative',
```
Comment (IoanSava) [inline cryk-ui/src/pages/Cryptocurrencies.tsx]: Use double quotes everywhere

File: cryk-ui/src/pages/Cryptocurrencies.tsx [hunk 6]
```diff
@@ -133,10 +194,40 @@ export default function Cryptocurrencies() {
     const [totalCoinsStats, setTotalCoinsStats] = useState<{ name: string, value: number }[]>([]);
     const [blockTimeStats, setBlockTimeStats] = useState<{ name: string, value: number }[]>([]);
 
+    const [searchItems, setSearchItems] = useState<string[]>([]);
+    const classes = useStyles();
+    const [value, setValue] = React.useState("");
```
Comment (IoanSava) [inline cryk-ui/src/pages/Cryptocurrencies.tsx]: Rename this state name

File: cryk-ui/src/pages/Cryptocurrencies.tsx [hunk 7]
```diff
@@ -133,10 +194,40 @@ export default function Cryptocurrencies() {
...
+
+    const handleKeyDown = (event: any) => {
+        if (["Enter", "Tab", ","].includes(event.key)) {
+            event.preventDefault();
+
+            var searchItem = value?.trim();
```
Comment (IoanSava) [inline cryk-ui/src/pages/Cryptocurrencies.tsx]: Don't use `var`. Use `let` instead

File: cryk-ui/src/pages/Cryptocurrencies.tsx [hunk 8]
```diff
@@ -133,10 +194,40 @@ export default function Cryptocurrencies() {
...
+            if (searchItem) {
+                setSearchItems([...searchItems, searchItem]);
+                setValue(event.target.value);
+                setSearchTextValue([...searchItems, searchItem]);
+            }
+            setValue("");
+        }
+    };
+
+    const handleDelete = (searchItem: string) => (e: any) => {
```
Comment (IoanSava) [inline cryk-ui/src/pages/Cryptocurrencies.tsx]: `e` => `event`

File: cryk-ui/src/pages/Cryptocurrencies.tsx [hunk 9]
```diff
@@ -192,14 +315,15 @@ export default function Cryptocurrencies() {
...
-                        onChange={(event, value) => setCurrentPage(value)} />
+                    {data.cryptocurrenciesInfo.totalCount !== 0 ?
+                        <Pagination className="pagination"
+                            count={totalNumberOfPages}
+                            color="primary"
+                            size="large"
+                            page={currentPage}
+                            variant="outlined"
+                            onChange={(event, value) => setCurrentPage(value)} />
+                        : null}
```
Comment (IoanSava) [inline cryk-ui/src/pages/Cryptocurrencies.tsx]: We don't need `: null`. `{data.cryptocurrenciesInfo.totalCount !== 0 ? <Pagination ... />}` is enough

File: cryk-ui/src/pages/Cryptocurrencies.tsx [hunk 10]
```diff
@@ -178,7 +269,39 @@ export default function Cryptocurrencies() {
...
+                                onDelete={handleDelete(item)}
+                                className={classes.chip}
+                                key={index}
+                            />
+                        ))}
+                    </div>
+
                     <div className="cards-container">
+                        {data.cryptocurrenciesInfo.totalCount === 0 ?
+                            <h2 className="no-cryptos">There are no cryptocurrencies matching the filters.</h2> : null}
```
Comment (IoanSava) [inline cryk-ui/src/pages/Cryptocurrencies.tsx]: We don't need `: null`

File: cryk-ui/src/pages/Cryptocurrencies.tsx (hunk 9 above)
Comment (DianaCrainic) [reply cryk-ui/src/pages/Cryptocurrencies.tsx]: It doesn't work like this

**Review by IoanSava — APPROVED** [review]

**Pull request merged**